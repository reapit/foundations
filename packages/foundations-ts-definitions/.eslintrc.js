module.exports = {
  env: {
    browser: true,
    es6: true,
    node: true,
    amd: true,
    jest: true,
  },
  extends: ['eslint:recommended'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 2018,
    sourceType: 'module',
  },
}
