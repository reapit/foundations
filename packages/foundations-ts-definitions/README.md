# Foundations TS Definitions:

Automated TypeScript definitions for the Foundations API Platform. Automatically generates up to date TypeScript type definitions from Platform API swagger documentation. 

For detailed documentation [visit here](https://foundations-documentation.reapit.cloud/app-development/foundations-ts-defintions).
