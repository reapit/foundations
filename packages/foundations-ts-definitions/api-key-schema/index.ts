export enum ApiKeyEntityType {
  PAYMENT = 'payment',
  DEPLOYMENT = 'deployment',
}

export interface ApiKeyInterface {
  id?: string
  apiKey?: string
  keyCreatedAt?: string
  keyExpiresAt?: string
  developerId?: string
  entityId?: string
  entityType?: ApiKeyEntityType
}
