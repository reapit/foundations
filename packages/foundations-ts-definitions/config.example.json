{
  "MARKETPLACE_API_BASE_URL": "",
  "MARKETPLACE_API_KEY": "",
  "PLATFORM_API_BASE_URL": ""
}
