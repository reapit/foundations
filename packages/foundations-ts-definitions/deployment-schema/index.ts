import { CodeBuild } from 'aws-sdk'

export interface PipelineModelInterface {
  id?: string
  name?: string
  organisationId?: string
  created?: string
  modified?: string
  developerId?: string
  appType?: AppTypeEnum
  buildCommand?: string
  packageManager?: PackageManagerEnum
  repository?: string
  outDir?: string
}

export interface PipelineRunnerModelInterface {
  id?: string
  pipelineId?: string
  created?: string
  modified?: string
  appType?: AppTypeEnum
  buildCommand?: string
  packageManager?: PackageManagerEnum
  tasks?: TaskModelInterface[]
  buildStatus?: CodeBuild.StatusType
}

export interface TaskModelInterface {
  id?: string
  pipelineRunnerId?: string
  created?: string
  modified?: string
  functionName?: CodeBuild.BuildPhaseType
  buildStatus?: CodeBuild.StatusType
}

export enum PackageManagerEnum {
  NPM = 'npm',
  YARN = 'yarn',
}

export enum AppTypeEnum {
  NODE = 'node',
  REACT = 'react',
}
