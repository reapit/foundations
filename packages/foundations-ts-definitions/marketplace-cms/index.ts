export interface AppsBrowseConfigItemContentInterface {
  brandColour?: string
  strapline?: string
  imageUrl?: string
  title?: string
  iconName?: string
}

export interface AppsBrowseConfigItemFiltersInterface {
  developerId?: string
  category?: string[]
  desktopIntegrationTypeId?: string[]
  id?: string[]
  appName?: string
  isFeatured?: boolean
  isFree?: boolean
  searchTerm?: string
}

export enum AppsBrowseConfigEnum {
  FEATURED_HERO = 'featuredHeroApps',
  HERO = 'heroApps',
  FILTERS = 'appsFilters',
  FEATURED = 'featuredApps',
  SIMPLE = 'simpleApps',
}

export interface AppBrowseLiveDataInterface {
  timeFrom?: string
  timeTo?: string
  isLive?: boolean
}

export interface AppsBrowseConfigItemInterface {
  id?: string
  index?: number
  filters?: AppsBrowseConfigItemFiltersInterface
  content?: AppsBrowseConfigItemContentInterface
  configType: AppsBrowseConfigEnum
  live: AppBrowseLiveDataInterface
}
