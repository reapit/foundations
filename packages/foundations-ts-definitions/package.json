{
  "name": "@reapit/foundations-ts-definitions",
  "version": "0.0.70",
  "license": "MIT",
  "files": [
    "types/*"
  ],
  "main": "./types/index.ts",
  "scripts": {
    "fetch-definition": "node '../../scripts/foundations-ts-definitions/fetch-definition.js'",
    "handle-cron-job": "node '../../scripts/foundations-ts-definitions/handle-cronjob.js'",
    "release:prod": "node ../../scripts/foundations-ts-definitions/release-npm.js foundations-ts-definitions",
    "test:update-badges": "echo 'not implmented'"
  },
  "lint-staged": {
    "src/**/*.{js,jsx,ts,tsx,json,css,scss,md}": [
      "prettier --write",
      "git add"
    ]
  },
  "dependencies": {},
  "peerDependencies": {}
}