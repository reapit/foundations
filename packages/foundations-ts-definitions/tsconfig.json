{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": "./"
  },
  "include": ["types"],
  "exclude": ["node_modules"]
}
