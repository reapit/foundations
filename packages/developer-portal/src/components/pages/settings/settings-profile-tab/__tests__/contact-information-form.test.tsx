import React from 'react'
import { mount } from 'enzyme'
import {
  defaultInitialValues,
  generateInitialValues,
  ContactInformationForm,
  ContactInformationValues,
  handleSubmitContactInformation,
} from '../contact-information-form'
import appState from '@/reducers/__stubs__/app-state'
import { MemberModel } from '@reapit/foundations-ts-definitions'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

const dispatch = jest.fn()
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(() => dispatch),
  useSelector: jest.fn(() => jest.fn()),
}))

const currentMemberInfo: MemberModel | null = appState.currentMember.data

const valuesMock: ContactInformationValues = {
  name: 'name',
  jobTitle: 'jobTitle',
}

describe('ContactInformationForm', () => {
  const mockStore = configureStore()
  const store = mockStore(appState)

  it('should match snapshot', () => {
    const wrapper = mount(
      <Provider store={store}>
        <ContactInformationForm />
      </Provider>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('handleSubmitContactInformation', () => {
  it('should run correctly', () => {
    const updateCurrentMemberInformation = jest.fn()
    const fn = handleSubmitContactInformation(updateCurrentMemberInformation)
    fn(valuesMock)
    expect(updateCurrentMemberInformation).toHaveBeenCalledWith(valuesMock)
  })
})

describe('generateInitialValues', () => {
  it('should return correctly', () => {
    const result = generateInitialValues({
      currentMemberInfo,
      defaultInitialValues,
    })
    const { name, jobTitle } = currentMemberInfo as MemberModel
    const expectedResult = {
      name,
      jobTitle,
    }
    expect(result).toEqual(expectedResult)
  })

  it('should return correctly with currentMemberInfo null', () => {
    const result = generateInitialValues({
      currentMemberInfo: null,
      defaultInitialValues,
    })
    expect(result).toEqual(defaultInitialValues)
  })
})
