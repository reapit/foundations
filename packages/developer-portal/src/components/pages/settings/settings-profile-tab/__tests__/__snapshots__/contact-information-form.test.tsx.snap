// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInformationForm should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Component>
    <Formik
      initialValues={
        Object {
          "jobTitle": "",
          "name": "mockConstructor",
        }
      }
      onSubmit={[Function]}
      validationSchema={
        ObjectSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_defaultDefault": [Function],
          "_deps": Array [],
          "_excludedEdges": Array [],
          "_exclusive": Object {},
          "_mutate": undefined,
          "_nodes": Array [
            "jobTitle",
            "name",
          ],
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "object",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "fields": Object {
            "jobTitle": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "matches": false,
                "max": true,
                "required": true,
                "trim": false,
              },
              "_mutate": undefined,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
                [Function],
              ],
              "type": "string",
            },
            "name": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "matches": false,
                "max": true,
                "required": true,
                "trim": false,
              },
              "_mutate": undefined,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
                [Function],
              ],
              "type": "string",
            },
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
          "type": "object",
        }
      }
    >
      <Component>
        <div
          className=""
        >
          <Form>
            <form
              action="#"
              onReset={[Function]}
              onSubmit={[Function]}
            >
              <mock-styled.div>
                <Component>
                  <div
                    className=""
                  >
                    Contact Information
                  </div>
                </Component>
                <Component>
                  <div
                    className=""
                  >
                    Please use the fields below to edit your contact information
                  </div>
                </Component>
                <Component>
                  <div
                    className=""
                    data-test=""
                  >
                    <Component>
                      <div
                        className=""
                        data-test=""
                      >
                        <Component
                          dataTest="name"
                          disabled={[MockFunction]}
                          id="name"
                          labelText="Full Name"
                          name="name"
                          type="text"
                        >
                          <Field
                            name="name"
                            validate={null}
                          >
                            <div
                              className=""
                            >
                              <div
                                className="control "
                              >
                                <label
                                  className="label inline-block "
                                  htmlFor="name"
                                >
                                  Full Name
                                </label>
                                <input
                                  className=""
                                  data-test="name"
                                  disabled={[MockFunction]}
                                  id="name"
                                  name="name"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  placeholder=""
                                  type="text"
                                  value="mockConstructor"
                                />
                              </div>
                            </div>
                          </Field>
                        </Component>
                      </div>
                    </Component>
                    <Component>
                      <div
                        className=""
                        data-test=""
                      >
                        <Component
                          dataTest="job-title"
                          disabled={[MockFunction]}
                          id="jobTitle"
                          labelText="Job Title"
                          name="jobTitle"
                          type="text"
                        >
                          <Field
                            name="jobTitle"
                            validate={null}
                          >
                            <div
                              className=""
                            >
                              <div
                                className="control "
                              >
                                <label
                                  className="label inline-block "
                                  htmlFor="jobTitle"
                                >
                                  Job Title
                                </label>
                                <input
                                  className=""
                                  data-test="job-title"
                                  disabled={[MockFunction]}
                                  id="jobTitle"
                                  name="jobTitle"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  placeholder=""
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </Field>
                        </Component>
                      </div>
                    </Component>
                  </div>
                </Component>
                <Component>
                  <div
                    className=""
                    data-test=""
                  >
                    <Component
                      dataTest="save-changes"
                      disabled={false}
                      loading={[MockFunction]}
                      type="submit"
                      variant="primary"
                    >
                      <button
                        className="button  is-primary  is-loading"
                        data-test="save-changes"
                        disabled={false}
                        type="submit"
                      >
                        Save Changes
                      </button>
                    </Component>
                  </div>
                </Component>
              </mock-styled.div>
            </form>
          </Form>
        </div>
      </Component>
    </Formik>
  </Component>
</Provider>
`;
